lib/reservation.rb

```diff
@@ -0,0 +1,67 @@
+require 'date'
+require 'pry'
+
+require_relative 'room'
+
+module HotelManager
+	class Reservation
+
+		attr_reader :id, :customer_id, :start_date, :end_date, :room_cost, :rooms, :room_ids
+
+		# Room default cost of $200
+		def initialize id: , 
+									customer_id: , 
+									start_date: , 
+									end_date: , 
+									room_cost: 200, 
+									rooms: nil,
+									room_ids: nil
+
+			@id = id 
+			@customer_id = customer_id
+			@start_date = start_date
+			@end_date = end_date
+			@room_cost = room_cost
+			@rooms = rooms 
+			@room_ids = room_ids
+
+			validate_date			
+			@rooms.nil? ? validate_room(@room_ids) : validate_room(@rooms)
+		end
+
+		def validate_room(attribute)
```

Comment:
Consider giving this method a name like `validate_block` or adding a comment to increase readability.